internal/controller/spinapp_controller.go

```diff
@@ -461,11 +461,17 @@ func constructDeployment(ctx context.Context, app *spinv1alpha1.SpinApp, config
 			ReadinessProbe: readinessProbe,
 		}
 	} else if config.SpinImage != nil {
+		args := []string{"up", "--listen", fmt.Sprintf("0.0.0.0:%d", spinapp.DefaultHTTPPort), "-f", app.Spec.Image, "--runtime-config-file", "/runtime-config.toml"}
+		if app.Spec.Components != nil {
+			for _, component := range app.Spec.Components {
+				args = append(args, "--component-id", component)
```

Comment:
What happens if this feature is used with a version of the shim that doesn't support it?

Reply:
The spin pod will throw an error for a disallowed flag. It should not be used with a spin version < 3.0